app_legacy/Helpers/render/user.php

```diff
@@ -147,23 +147,52 @@ function renderUserCard(string|array $user): string
     return $tooltip;
 }
 
+function getCompletedAndIncompletedSetsCounts(array $userCompletedGamesList): array
+{
+    $completedSetsCount = 0;
+    $incompletedSetsCount = 0;
+
+    $numItems = count($userCompletedGamesList);
+    for ($i = 0; $i < $numItems; $i++) {
```

Comment:
Consider using foreach:
```
foreach ($userCompletedGamesList as $game) {
    $nextMaxPossible = $game['MaxPossible']
    ....
```

Avoids doing the count, and two indexed lookups for every iteration.

Reply:
That does simplify it quite a bit. Also resolved in https://github.com/RetroAchievements/RAWeb/pull/1492/commits/5353618147739ece716f4f3e6ef5991ddf1d9098.